driver-sync/src/main/com/mongodb/client/ListCollectionsIterable.java

```diff
@@ -39,6 +40,21 @@ public interface ListCollectionsIterable<TResult> extends MongoIterable<TResult>
      */
     ListCollectionsIterable<TResult> filter(@Nullable Bson filter);
 
+    /**
+     * Sets the {@code authorizedCollections} field of the {@code listCollections} command.
+     * This method is ignored if called on a {@link ListCollectionsIterable} obtained not via any of the
+     * {@link MongoDatabase#listCollectionNames() MongoDatabase.listCollectionNames} methods.
```

Comment:
There are alternatives to ignoring the option, as described in the "API design" section of the PR description.